### ISSUE TITLE
invoke function launches new lambdas after ~300 seconds

### ISSUE THREAD

**Issue by kylegibson-rldatix** [Miserlou/Zappa#1691]
## Context

I have a zappa environment configured for a timeout of 900 seconds. I used `zappa invoke <stage> function` to invoke a lambda instance running a particular function, one that takes over 400 seconds to complete. I was tailing the logs during the process.

## Expected Behavior

Only 1 lambda instance should be launched.

## Actual Behavior

After about 300 seconds, another lambda instance was launched. The logs clearly have "Zappa Event: {'command': '...'}" twice, even though I only ran `invoke` once. 

## Possible Fix

https://github.com/boto/boto3/issues/1104

## Your Environment
* Zappa version used: 0.47
* Operating System and Python version: ubuntu 16.04, pytohn 3.6.5

**Comment by Lexy2** [comment]
I guess the reason is that after about 300 seconds zappa.keep_warm_callback wants to kick in, but since the previous LambdaHandler instance is still running, the event object is spoiled.
As a workaround, you could remove the warm_callback (as your function runs longer than that) or decrease the rate of warm_callback invocation to some 30 minutes.

**Comment by kylegibson** [comment]
I don't believe it was the keep warm callback. I have that disabled, and the logs did not show it being started.

**Comment by kylegibson** [comment]
I found that I could avoid this problem if I invoked functions that subsequently called another function that was `@task` decorated. In this case, the invoke will return as soon as the first function finishes, but then the task continues on its own.
